dms/models/ir_attachment.py

```diff
@@ -0,0 +1,85 @@
+import re
+
+from odoo import api, models
+
+
+class IrAttachment(models.Model):
+
+    _inherit = "ir.attachment"
+
+    def _dms_file_name(self, directory_id, file_name):
+        count = 1
+        for file in directory_id.file_ids:
+
+            dms_file_name = re.sub(r" \([0-9]*\)", "", file.name)
+            if dms_file_name == file_name:
+                count += 1
+
+        file, ext = re.split(r"\.", file_name)
+        return file + " ({}).".format(count) + ext if count > 1 else file_name
+
+    def _create_record_dir(self, storage_id, attachment_id):
+        dms_field = self.env["ir.module.module"].search([("name", "=", "dms_field")])
+
+        directory_fields = {
+            "name": attachment_id.res_name.replace("/", "-"),
+            "is_root_directory": True,
+            "parent_id": False,
```

Comment:
Wouldnt it be better to add all these directories under a common model directory `Contacts Root` for example instead of making every record root directory?